src/dasi/core/Retriever.cc

```diff
@@ -0,0 +1,69 @@
+
+#include "dasi/api/ReadHandle.h"
+#include "dasi/api/Query.h"
+#include "dasi/api/RetrieveResult.h"
+
+#include "dasi/core/Retriever.h"
+
+using dasi::api::Config;
+
+namespace dasi::core {
+
+//----------------------------------------------------------------------------------------------------------------------
+
+class RetrieveVisitor {
+
+public: // methods
+
+    RetrieveVisitor(Retriever& parent, api::RetrieveResult& res) :
+        parent_(parent),
+        res_(res) {}
+
+    void thirdLevel(SplitReferenceKey& key) {
+        std::cout << "Third level in retrieve!!!" << std::endl;
+        DB& db = parent_.database(key[0]);
+        auto result = db.retrieve(key);
+        api::Key key2;
```

Comment:
1. If we want to be able to convert a SplitReferenceKey to a Key, then this should be a member function of SplitReferenceKey
2. Do we need to do this? Can we not continue using a SplitReferenceKey to refer to the retrieved data. That way carries more information anyway.

Reply:
Seee src/dasi/api/SplitKey.h - it already even has a constructor that will take a SplitReferenceKey.